src/test/shell/integration/target_compatible_with_test.sh

```diff
@@ -1082,4 +1082,164 @@ function test_aquery_incompatible_target() {
   expect_log "target platform (//target_skipping:foo3_platform) didn't satisfy constraint //target_skipping:foo1"
 }
 
+# Use aspects to interact with incompatible targets and validate the behaviour.
+function test_aspect_skipping() {
+  cat >> target_skipping/BUILD <<EOF
+load(":defs.bzl", "basic_rule", "rule_with_aspect")
+
+basic_rule(
+    name = "basic_universal_target",
+)
```

Comment:
I added this target here that is depended on by `basic_foo3_target`. The idea being that if `basic_foo3_target` is incompatible, then aspects should never reach `basic_universal_target`.

As far as I can tell, this works as expected:
* The `print()` statement in the aspect never gets executed for `basic_universal_target` if `basic_foo3_target` is incompatible.
* The `print()` statement for `basic_universal_target` _does_ get executed when `basic_foo3_target` is compatible.

The test now checks for the debug message on this target.

Reply:
That sounds great, thanks.

Would you mind comment-documenting basic_universal_target to clarify the point you made? That this is to cover the incompatible -> compatible dependency order?

And maybe a similar comment on the target that shows the opposite?

Reply:
That's a good idea!

Done